app/views/users/index.html.erb

```
@@ -1,3 +1,4 @@
+<% if @login_user %>
 <h3>List of Users</h3>
 <table>
```

While this technique of checking the login user before rendering the view template prevents users from seeing pages they don't have access to, **it does not provide security for your site**. Because the restrictions are in the view not the controller, any action that bypasses the view (such as creating or deleting a work) will not be restricted to logged in users. This means that a malicious user could use a program like curl or Postman to cause problems.

Even for `GET` requests, this strategy has some drawbacks:
- The code is repeated everywhere, and needs to be made DRY
- When adding a new page it would be easy to forget to add this check
- It will not work for APIs, which have no views
- The error is not reported to the user via flash
- There is no way to associate an HTTP status code with the failure

Instead, you should use a controller filter on the `ApplicationController` to restrict access to actions, and turn the filter off for actions that anyone can take.